_posts/2023-03-24-prospero.adoc

```diff
@@ -0,0 +1,308 @@
+---
+layout: post
+title:  "An Introduction To The Prospero Usage"
+date:   2023-03-24
+tags:   prospero galleon wildfly
+author: liweinan
+description: An introduction to the Prospero usage.
+---
+
+= An Introduction To The Prospero Usage
+
+`Prospero` is a tool provided by the WildFly community:
+
+* https://github.com/wildfly-extras/prospero[GitHub - wildfly-extras/prospero: Prospero is a tool combining Galleon feature packs and wildfly-channels to provision and update Wildfly server.]
+
+The purpose of the project can be quoted from the README of the above project link:
+
+____
+Prospero is a tool combining Galleon feature packs and wildfly-channels to provision and update Wildfly server.
+____
+
+Briefly speaking, you can use the tool to do the version management of your WildFly installation.
+
+[NOTE]
+====
+The `wildfly-channels` projects defines the `Channel`, which allow separation of artifact versioning from the Galleon feature pack, allowing artifact versions to be managed independently. In the project, it contains a spec doc that describes the `Channel` model:
+
+* https://github.com/wildfly-extras/wildfly-channel/blob/main/doc/spec.adoc[wildfly-channel/spec.adoc at main · wildfly-extras/wildfly-channel · GitHub]
+====
+
+In the next section, let's see the basic usage of `prospero-cli`.
+
+== Basic Usage Of The Prospero CLI Tool
+
+To use the tool, firstly you can clone the project to your local environment:
+
+[,bash]
+----
+$ git clone https://github.com/wildfly-extras/prospero.git
+----
+
+Then you can enter the project directory, and build the project as mentioned in the project README:
+
+[,bash]
+----
+$ mvn clean install
+----
+
+After the project is built, then we can use the built tool to install a provisioned WildFly locally. The tool uses the `wildfly-channel` config files(which are called `Wildfly Channels` and `Wildfly Channel Manifests`) to manage the components versions. And the project provides some example configuration files already in its `examples` directory:
+
+[,bash]
+----
+➤ ls examples
+README.md                           wildfly-27.0.0.Alpha2-manifest.yaml
+wildfly-26.0.0.Final-channel.yaml   wildfly-core-channel.yaml
+wildfly-26.0.0.Final-manifest.yaml  wildfly-core-manifest.yaml
+wildfly-27.0.0.Alpha2-channel.yaml
+----
+
+To use one of the above manifest files, here is an example command:
+
+[,bash]
+----
+$ ./prospero install --fpl=wildfly --dir=wfly-26 --channel=examples/wildfly-26.0.0.Final-channel.yaml
```

Comment:
just a note - I'm planning to [split the `--fpl` argument into `--profile` and `--fpl`](https://github.com/wildfly-extras/prospero/pull/334) to make the usage a bit less confusing.

It might be good to incorporate this change in the blog - or can we update it after it's published?

Reply:
@spyrkob Both options are fine to me :D btw do what's the differences between the `--profile` and `--fpl`? And where are the profiles come from?

Reply:
@spyrkob Thanks for the explanation! I guess this an important feature split so I'll update the blog post on this part :D

Reply:
@liweinan FYI I just released a new version of Prospero with the `--spl`/`--profile` split.

Reply:
@spyrkob Cool! I'll play with the new feature and if I have questions I'll ask you questions :D

And then I'll update this article accordingly.